backend/services/cache.ts

```diff
@@ -0,0 +1,21 @@
+import { contractDatabaseRef } from './db';
+import { getIPFSData } from './ipfs';
+
+export const getIPFSfromCache = async (cid: string) => {
+  const ref = contractDatabaseRef(`ipfs/${cid}`);
+
+  const res = (await ref.get()).val();
+
+  if (!res) {
+    const metadata = await (await getIPFSData(cid)).json();
+
```

Comment:
This needs the `transformIPFStoJob` method so that the frontend can present the labels properly.